src/main/kotlin/com/launchdarkly/intellij/toolwindow/FlagNodeViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.launchdarkly.intellij.toolwindow
+
+import com.launchdarkly.api.model.FeatureFlag
+import com.launchdarkly.api.model.FeatureFlags
+import com.launchdarkly.intellij.LDIcons
+import com.launchdarkly.intellij.featurestore.FlagConfiguration
+
+class FlagNodeViewModel(
+    val flag: FeatureFlag,
+    val flags: FeatureFlags,
+    val flagConfig: FlagConfiguration?,
+) {
+    val description = flag.description
+    val isEnabled = flagConfig?.on == true
+    val isDisconnected = flagConfig == null
+    val targets = flagConfig?.targets ?: listOf()
+    val prereqFlags = flagConfig?.prerequisites ?: listOf()
+    val numRules = flagConfig?.rules?.size ?: 0
+    val hasFallthrough = flagConfig?.fallthrough != null
+    val hasOffVariation = flagConfig?.offVariation != null
+    private val offVariationIndex = flagConfig?.offVariation as? Int
+    val offVariation = when {
+        offVariationIndex != null -> flag.variations[offVariationIndex].name ?: flag.variations[offVariationIndex].value
+        else -> null
+    }
+    val flagLabel = flag.name ?: flag.key
+    val icon = when {
+        isDisconnected -> LDIcons.TOGGLE_DISCONNECTED
+        isEnabled -> LDIcons.TOGGLE_ON
+        else -> LDIcons.TOGGLE_OFF
+    }
```

Comment:
nit

minor, but i believe there's a state where `isEnabled=null` (it should only happen when `isDisconnected=true`), but for clarity sake I think it might be better to...

```suggestion
    val icon = when {
        isEnabled -> LDIcons.TOGGLE_ON
        isEnabled == false -> LDIcons.TOGGLE_OFF
        else -> LDIcons.TOGGLE_DISCONNECTED
    }
```

feel free to veto, but i like the fallback being disconnected

Reply:
`isEnabled` is just Boolean, so it'll default to false if `flagConfig` is `null`, but I agree that having the fallback as disconnected is a better model but I don't think the way it's set up now it could ever be the fallback. Alternatively I can make `isEnabled` `Boolean?` so we can cover this state?

Reply:
fixed -thanks!